examples/6tisch/simple-node/Makefile

```diff
@@ -15,28 +15,45 @@ MAKE_WITH_PERIODIC_ROUTES_PRINT ?= 0
 MAKE_WITH_STORING_ROUTING ?= 0
 # Orchestra link-based rule? (Works only if Orchestra & storing mode routing is enabled)
 MAKE_WITH_LINK_BASED_ORCHESTRA ?= 0
+# Use the Orchestra root rule?
+MAKE_WITH_ORCHESTRA_ROOT_RULE ?= 0
 
 MAKE_MAC = MAKE_MAC_TSCH
 
 include $(CONTIKI)/Makefile.dir-variables
 MODULES += $(CONTIKI_NG_SERVICES_DIR)/shell
 
+ORCHESTRA_EXTRA_RULES = &unicast_per_neighbor_rpl_ns
+
 ifeq ($(MAKE_WITH_ORCHESTRA),1)
   MODULES += $(CONTIKI_NG_SERVICES_DIR)/orchestra
 
   ifeq ($(MAKE_WITH_STORING_ROUTING),1)
     ifeq ($(MAKE_WITH_LINK_BASED_ORCHESTRA),1)
       # enable the `link_based` rule
-      CFLAGS += -DORCHESTRA_CONF_RULES="{&eb_per_time_source,&unicast_per_neighbor_link_based,&default_common}"
+      ORCHESTRA_EXTRA_RULES = &unicast_per_neighbor_link_based
     else
       # enable the `rpl_storing` rule
-      CFLAGS += -DORCHESTRA_CONF_RULES="{&eb_per_time_source,&unicast_per_neighbor_rpl_storing,&default_common}"
+      ORCHESTRA_EXTRA_RULES = &unicast_per_neighbor_rpl_storing
+    endif
+
+    ifeq ($(MAKE_WITH_ORCHESTRA_ROOT_RULE),1)
+      # add the root rule
+      ORCHESTRA_EXTRA_RULES +=,&special_for_root
     endif
   else
     ifeq ($(MAKE_WITH_LINK_BASED_ORCHESTRA),1)
-      $(error "Inconsistent configuration")
+      $(error "Inconsistent configuration: link-based Orchestra requires routing info")
+    endif
+
+    ifeq ($(MAKE_WITH_ORCHESTRA_ROOT_RULE),1)
+     $(error "Inconsistent configuration: NS rule and root rule conflicts!")
```

Comment:
Wouldn't it be possible to run the root rule along with NS, at least in theory?
Send following receiver schedule, but if receiver is root send anytime?

Reply:
Good comment - I had to think for a while to remember what the issue was.
I think the issue is in the prioritization of the different rules - NS rule is prioritized over the root rule, because it has a lower slotframe handle. And the NS rule all slots potentially active, with all-except-one slots marked as Tx. This leads to the NS slot is always being selected over the root rule slot. Meaning that the root rule never get used for transmission, and the queue of packets towards the root just fills up and never gets cleared.

Reply:
For the NS rule, there is a related PR: https://github.com/contiki-ng/contiki-ng/pull/1520
It's basically ready from my side, but I'm waiting for this one to be merged before moving it out of the draft status, because its based on top of this one. It would be great if you could also look there and give quick feedback about the idea.